client/executor/wasmtime/src/instance_wrapper.rs

```diff
@@ -377,3 +404,14 @@ impl InstanceWrapper {
 		&mut self.store
 	}
 }
+
+#[test]
+fn decommit_works() {
+	let engine = wasmtime::Engine::default();
+	let code = wat::parse_str("(module (memory (export \"memory\") 1 4))").unwrap();
+	let module = Module::new(&engine, code).unwrap();
+	let mut wrapper = InstanceWrapper::new::<()>(&module, 2, true, None).unwrap();
+	unsafe { *wrapper.memory.data_ptr(&wrapper.store) = 42 };
```

Comment:
I am just paranoid :)

```suggestion
	unsafe { *wrapper.memory.data_ptr(&wrapper.store) = 42 };
	assert_eq!(unsafe { *wrapper.memory.data_ptr(&wrapper.store) }, 42);
```